docs/intro/quickstart/access.rst

```diff
@@ -0,0 +1,423 @@
+    +   single optional global current_user := (
+    +     select AccessToken filter .id = global access_token
+    +   ).user;
+    +
+    +   type User {
+    +     required name: str;
+    +   }
+    +
+    +   type AccessToken {
+    +     required user: User;
+    +   }
+    +
+        type Deck {
+          required name: str;
+          description: str;
+    +
+    +     creator: User;
+
+          cards := (select .<deck[is Card] order by .order);
+    +
+    +     access policy creator_has_full_access
+    +       allow all
+    +       using (
+    +         .creator ?= global current_user
+    +       );
+        };
+
+        type Card {
+          required order: int64;
+          required front: str;
+          required back: str;
+
+          required deck: Deck;
+    +
+    +     access policy deck_creator_has_full_access
+    +       allow all
+    +       using (
+    +         .deck.creator ?= global current_user
+    +       );
+        }
+      }
+
+.. edb:split-section::
+
+  We will create this migration which will also trigger the query builder to be regenerated.
+
+  .. code-block:: sh
+
+    $ npx gel migration create
+    did you create global 'default::access_token'? [y,n,l,c,b,s,q,?]
+    > y
+    did you create object type 'default::User'? [y,n,l,c,b,s,q,?]
+    > y
+    did you create object type 'default::AccessToken'? [y,n,l,c,b,s,q,?]
+    > y
+    did you create global 'default::current_user'? [y,n,l,c,b,s,q,?]
+    > y
+    did you alter object type 'default::Deck'? [y,n,l,c,b,s,q,?]
+    > y
+    did you create access policy 'deck_creator_has_full_access' of object type 'default::Card'? [y,n,l,c,b,s,q,?]
+    > y
+    Created /home/strinh/projects/flashcards/dbschema/migrations/00003-m1solvt.edgeql, id: m1solvta35uzsbs4axzqmkwfx7zatjtkozpr43cjs56fp75qzbrg5q
+
+    $ npx gel migrate
+    Applying m1solvta35uzsbs4axzqmkwfx7zatjtkozpr43cjs56fp75qzbrg5q (00003-m1solvt.edgeql)
+    ... parsed
+    ... applied
+    Generating query builder...
+    Detected tsconfig.json, generating TypeScript files.
+    To override this, use the --target flag.
+    Run `npx @gel/generate --help` for full options.
+    Introspecting database schema...
+    Generating runtime spec...
+    Generating cast maps...
+    Generating scalars...
+    Generating object types...
+    Generating function types...
+    Generating operators...
+    Generating set impl...
+    Generating globals...
+    Generating index...
+    Writing files to ./dbschema/edgeql-js
+    Generation complete! 🤘
+
+.. edb:split-section::
+
+  Let's create a page for creating a new user and getting an access token. Let's start by creating the query to create a new user which will return the ``AccessToken.id`` which we will use as the access token itself. We will save this access token in a cookie so that we can authenticate requests in other server actions and route handlers.
+
+  .. tabs::
+
+    .. code-tab:: typescript
+      :caption: app/signup/actions.ts
```

Comment:
It looks like the indentation here is too big (comparing to other code blocks):

![CleanShot 2025-02-11 at 15 51 29@2x](https://github.com/user-attachments/assets/711bdf27-eefc-429b-b797-aa913ebc7b1a)

Reply:
Yeah, I think _all_ of my non-diff code blocks are wrong. I wrongly assumed I could make the diff and non-diff code blocks have the same indentation and that the directive would normalize them, but it looks like I cannot. That makes it _very_ annoying to start with a code block and then switch to the same code in a diff 😅 . Will fix once it's all finished.